src/Mappers/InventoryMapper.php

```diff
@@ -34,85 +41,131 @@ public function getAvailableDate($mainVariationId) {
+     * @var AbstractConfigHelper $configHelper
      */
-    $warehouseSupplierRepository = pluginApp(WarehouseSupplierRepository::class);
-    $supplierId = null;
-    $warehouseId = null;
+    $configHelper = pluginApp(AbstractConfigHelper::class);
+    $stockBuffer = $configHelper->getStockBufferValue();
+    if ($variationStock->netStock > $stockBuffer) {
+      return $variationStock->netStock - $stockBuffer;
+    } else {
+      return 0;
+    }
+  }
+
+  /**
+   * Create an Inventory DTO for each Warehouse in the Variation data
+   * @param $variationData
+   *
+   * @return RequestDTO[]
+   */
+  public function createInventoryDTOsFromVariation($variationData)
+  {
+    /** @var LoggerContract $loggerContract */
+    $loggerContract = pluginApp(LoggerContract::class);
+
+    /** @var RequestDTO[] $inventoryDTOs */
+    $inventoryDTOs = [];
+
+    $supplierPartNumber = $this->getSupplierPartNumberFromVariation($variationData);
+
+    $mainVariationId = $variationData['id'];
+    $nextAvailableDate = $this->getAvailableDate($mainVariationId); // Pending. Need Item
+
+    $stockList = $variationData['stock'];
     foreach ($stockList as $stock) {
       $warehouseId = $stock['warehouseId'];
-      if (isset($warehouseId)) {
-        $warehouseSupplierMapping = $warehouseSupplierRepository->findByWarehouseId($warehouseId);
-        $supplierId = isset($warehouseSupplierMapping) ? $warehouseSupplierMapping->supplierId : null;
-        if (isset($supplierId)) {
-          $stock['supplierId'] = $supplierId;
-
-          return $stock;
-        }
+      if (!isset($warehouseId)) {
+        // we don't know the warehouse, so we can't figure out the supplier ID.
+        // Not an error, but unexpected.
+        $loggerContract->warning(
+          TranslationHelper::getLoggerKey(self::LOG_KEY_STOCK_MISSING_WAREHOUSE),
+          [
+            'additionalInfo' => ['variationID' => $mainVariationId,],
+            'method' => __METHOD__
+          ]
+        );
+        continue;
       }
+
+      // this is a 'mixed' value - might be null.
+      $supplierId = $this->getSupplierIDForWarehouseID($warehouseId);
+
+      if (!isset($supplierId) || $supplierId === 0 || $supplierId === '0') {
+        // no supplier assigned to this warehouse - NOT an error - we should NOT sync it
+        $loggerContract->debug(
+          TranslationHelper::getLoggerKey(self::LOG_KEY_NO_SUPPLIER_ID_ASSIGNED_TO_WAREHOUSE),
+          [
+            'additionalInfo' => ['warehouseID' => $warehouseId, 'variationID' => $mainVariationId,],
+            'method' => __METHOD__
+          ]
+        );
+        continue;
+      }
+
+      $dtoData = [
+        'supplierId' => $supplierId,
+        'supplierPartNumber' => $supplierPartNumber,
+        'quantityOnHand' => isset($stock['netStock']) ? $stock['netStock'] : null, // Avl Immediately. Net Stock.
+        'quantityOnOrder' => isset($stock['reservedStock']) ? $stock['reservedStock'] : null, // Already Ordered items.
```

Comment:
The term 'reserved' in this case means it's on order and not that this is stock explicitly reserved for Wayfair and no other retailer the supplier deals with?

Reply:
This is a Plentymarkets-side value, and is not Wayfair-specific.

> The integrated stock management system... can automatically reserve stock for listings or pending orders.
https://www.plentymarkets.co.uk/product/stock-management-statistics/